server.js

```diff
@@ -7,23 +7,61 @@ const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
+const Thought = mongoose.model('Thought', {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140
+  },
+  hearts: {
+    type: Number, 
+    default: 0
+  },
+  createdAt: {
+    type: Date,
+    default: Date.now
+  }
+})
+
+const port = process.env.PORT || 8081
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('welcome to the happy thougths api!')
+})
+
+app.get('/thoughts', async (req, res) => {
+  const thougths = await Thought.find().sort({createdAt: 'desc' }).limit(20).exec()  
+  res.json(thougths)
+})
+
+app.post('/thoughts', async (req, res) => {
+  const { message } = req.body 
+  const thought = new Thought({ message })
+
+  try {
+    const savedThought = await thought.save()
+    res.status(201).json(savedThought)
+  } catch (err) {
+  res.status(400).json({ message: 'could not post happy thought', errors:err.errors})
+ }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => { 
+  const thoughtId = req.params.thoughtId
+
+  try {
```

Comment:
Great Job Moa and congratulations on completing the blue level. The functionality work both on the front and the back end well. The code was clean and easy to read. 🥇 💯